src/logic/wallets/onboard.ts

```diff
@@ -45,28 +45,28 @@ const getOnboard = (chainId: ChainId): API => {
     networkName: getNetworkName(chainId),
     subscriptions: {
       wallet: async (wallet) => {
-        if (!wallet.provider) {
+        if (wallet.provider) {
+          setWeb3(wallet.provider)
+        } else {
           resetWeb3()
-          return
```

Comment:
No need to exit anymore?

Reply:
No, this was blocking the wallet subscription dispatching action. I discussed this directly with @francovenica.